sylt-parser/src/expression.rs

```diff
@@ -902,3 +902,151 @@ mod test {
+                b.pretty_print(f, indent + 1)?;
+            }
+            EK::And(a, b) => {
+                write!(f, "And\n")?;
+                a.pretty_print(f, indent + 1)?;
+                b.pretty_print(f, indent + 1)?;
+            }
+            EK::Or(a, b) => {
+                write!(f, "Or\n")?;
+                a.pretty_print(f, indent + 1)?;
+                b.pretty_print(f, indent + 1)?;
+            }
+            EK::Not(a) => {
+                write!(f, "Not\n")?;
+                a.pretty_print(f, indent + 1)?;
+            }
+            EK::Parenthesis(expr) => {
+                write!(f, "Paren\n")?;
+                expr.pretty_print(f, indent + 1)?;
+            }
+            EK::Duplicate(expr) => {
+                write!(f, "Duplicate\n")?;
+                expr.pretty_print(f, indent + 1)?;
+            }
+            EK::IfExpression { condition, pass, fail } => {
+                write!(f, "IfExpression\n")?;
+                write_indent(f, indent);
+                write!(f, "condition:\n")?;
+                condition.pretty_print(f, indent + 1)?;
+                write_indent(f, indent);
+                write!(f, "pass:\n")?;
+                pass.pretty_print(f, indent + 1)?;
+                write_indent(f, indent);
+                write!(f, "fail:\n")?;
+                fail.pretty_print(f, indent + 1)?;
+            }
+            EK::IfShort { lhs, condition, fail } => {
+                write!(f, "IfShort\n")?;
+                write_indent(f, indent);
+                write!(f, "lhs:\n")?;
+                lhs.pretty_print(f, indent + 1)?;
+                write_indent(f, indent);
+                write!(f, "pass:\n")?;
+                condition.pretty_print(f, indent + 1)?;
+                write_indent(f, indent);
+                write!(f, "fail:\n")?;
+                fail.pretty_print(f, indent + 1)?;
+            }
+            EK::Function { name, params, ret, body } => {
+                write!(f, "Fn {} ", name)?;
+                for (i, (name, ty)) in params.iter().enumerate() {
+                    if i != 0 { write!(f, ", ")?; }
+                    write!(f, "{}: {}", name.name, ty)?;
+                }
+                write!(f, " -> {}", ret);
+                write!(f, "\n");
+                body.pretty_print(f, indent + 1)?;
+            }
+            EK::Instance { blob, fields } => {
+                write!(f, "Instance ")?;
+                blob.pretty_print(f, indent + 1);
+                write!(f, "\n")?;
+                for (field, value) in fields.iter() {
+                    write_indent(f, indent)?;
+                    write!(f, ".{}:\n", field)?;
+                    value.pretty_print(f, indent + 1);
+                }
+            }
+            EK::Tuple(values) => {
+                write!(f, "Tuple\n")?;
+                for value in values.iter() {
+                    value.pretty_print(f, indent + 1);
+                }
```

Comment:
```suggestion
                values.iter().try_for_each(|value| value.pretty_print(f, indent + 1))
```
I think this works. It could probably be applied at other places in the repo.

Reply:
Very cool